test/unit/test_disk_tensor.py

```diff
@@ -128,32 +135,17 @@ def test_write_ones(self):
     out = reloaded.numpy()
     assert np.all(out == 1.)
 
-  def test_slice(self):
-    pathlib.Path(temp("dt3")).unlink(missing_ok=True)
-    Tensor.arange(10, device="CPU").to(f"disk:{temp('dt3')}").realize()
-
-    slice_me = Tensor.empty(10, device=f"disk:{temp('dt3')}")
-    print(slice_me)
-    is_3 = slice_me[3:4].cpu()
-    assert is_3.numpy()[0] == 3
-
-  def test_slice_2d(self):
-    pathlib.Path(temp("dt5")).unlink(missing_ok=True)
-    Tensor.arange(100, device="CPU").to(f"disk:{temp('dt5')}").realize()
-    slice_me = Tensor.empty(10, 10, device=f"disk:{temp('dt5')}")
-    tst = slice_me[1].numpy()
-    print(tst)
-    np.testing.assert_allclose(tst, np.arange(10, 20))
-
   def test_assign_slice(self):
-    pathlib.Path(temp("dt4")).unlink(missing_ok=True)
-    cc = Tensor.arange(10, device="CPU").to(f"disk:{temp('dt4')}").realize()
+    helper_test_disk_tensor("dt3", [0,1,2,3], lambda x: (x.__setitem__(slice(0,2), [13, 12]), x[0:2])[1], lambda x: x[0:2].assign([13, 12]))
+    helper_test_disk_tensor("dt4", [[0,1,2,3],[4,5,6,7]], lambda x: ((x.__setitem__(slice(0,1), [[13, 12, 11, 10]]), x[0:1])[1]), lambda x: x[0:1].assign([[13, 12, 11, 10]]))
+
+  def test_reshape(self):
+    helper_test_disk_tensor("dt5", [1,2,3,4,5], lambda x: x.reshape((1,5)))
+    helper_test_disk_tensor("dt6", [1,2,3,4], lambda x: x.reshape((2,2)))
 
-    #cc.assign(np.ones(10)).realize()
-    print(cc[3:5].numpy())
-    cc[3:5].assign([13, 12]).realize()
-    print(cc.numpy())
+  def test_permute(self):
+    helper_test_disk_tensor("dt7", [[1,2,3,4,5],[6,7,8,9,10]], lambda x: np.transpose(x, (0,1)), lambda x: x.permute(0,1))
```

Comment:
There's no way this works.

Reply:
yeah.. `as_strided` just worked in this case

Reply:
Oh...because that's a fake NOOP permute...